client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -151,7 +150,7 @@ const Checkout = React.createClass( {
 			return false;
 		}
 
-		if ( this.props.transaction.step.name === transactionStepTypes.SUBMITTING_WPCOM_REQUEST ) {
+		if ( includes( [ transactionStepTypes.SUBMITTING_WPCOM_REQUEST, transactionStepTypes.RECEIVED_WPCOM_RESPONSE ], this.props.transaction.step.name ) ) {
 			return false;
 		}
 
```

Comment:
Talking about comments, I was thinking of adding the following one here to explain the rest of the code in this method:

```
// Handles cases where the last item was removed by users from the cart, and we need to redirect them to a page that
// makes sense - according to the former content of their shopping cart
```